sdk/generic-sdk-core/generic-core/src/main/java/com/generic/core/http/models/HttpResponse.java

```diff
@@ -27,6 +52,9 @@ public class HttpResponse<T> implements Response<T> {
     private BinaryData body;
     private T value;
 
+    private Supplier<BinaryData> bodySupplier;
```

Comment:
Not a huge fan of having both `body` and `bodySupplier` as it's kind of confusing. Is this something we could push down into `BinaryData` itself as `BinaryData.fromBytes` could be considered the buffered format whereas `BinaryData.fromInputStream` could be considered the streaming format.

Reply:
I hadn't thought of things that way, I'll remove the `bodySupplier`.